src/rgw/rgw_rados.cc

```diff
@@ -2423,13 +2422,32 @@ int RGWRados::Bucket::update_bucket_id(const string& new_bucket_id)
  * marker: if filled in, begin the listing with this object.
  * end_marker: if filled in, end the listing with this object.
  * result: the objects are put in here.
- * common_prefixes: if delim is filled in, any matching prefixes are placed here.
- * is_truncated: if number of objects in the bucket is bigger than max, then truncated.
+ * common_prefixes: if delim is filled in, any matching prefixes are
+ * placed here.
+ * is_truncated: if number of objects in the bucket is bigger than
+ * max, then truncated.
  */
-int RGWRados::Bucket::List::list_objects_ordered(int64_t max,
-						 vector<rgw_bucket_dir_entry> *result,
-						 map<string, bool> *common_prefixes,
-						 bool *is_truncated)
+static inline std::string after_delim(boost::string_ref delim)
```

Comment:
can we stick to std::string_view from c++17? if it needs backporting to jewel, there's a boost::string_view whose interface is closer to std::string_view

Reply:
I didn't realize it had remove_prefix/suffix;  since it does, yeah, no problem